src/main/java/techreborn/enet/ElectricNetworkManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of TechReborn, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2020 TechReborn
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package techreborn.enet;
+
+import net.fabricmc.fabric.api.event.server.ServerTickCallback;
+import net.minecraft.server.MinecraftServer;
+import team.reborn.energy.EnergyTier;
+import techreborn.TechReborn;
+
+import java.util.HashSet;
+
+public final class ElectricNetworkManager implements ServerTickCallback {
+	public static ElectricNetworkManager INSTANCE;
+
+	private final HashSet<ElectricNetwork> electricNetworks = new HashSet<>();
+	private final HashSet<ElectricNetwork> pendingNetworks = new HashSet<>();
+
+	private ElectricNetworkManager() {
+		ServerTickCallback.EVENT.register(this::tick);
+	}
+
+	public static void init() {
+		if (INSTANCE == null) {
+			INSTANCE = new ElectricNetworkManager();
+		}
+	}
+
+	@Override
+	public void tick(MinecraftServer minecraftServer) {
+		electricNetworks.addAll(pendingNetworks);
+		pendingNetworks.clear();
```

Comment:
Whats with `pendingNetworks`? why can the not be added right into `electricNetworks`?